document_stores/elasticsearch/src/elasticsearch_haystack/document_store.py

```diff
@@ -237,16 +240,29 @@ def write_documents(self, documents: List[Document], policy: DuplicatePolicy = D
             index=self._index,
             raise_on_error=False,
         )
-        if errors and policy == DuplicatePolicy.FAIL:
-            # TODO: Handle errors in a better way, we're assuming that all errors
-            # are related to duplicate documents but that could be very well be wrong.
-
-            # mypy complains that `errors`` could be either `int` or a `list` of `dict`s.
-            # Since the type depends on the parameters passed to `helpers.bulk()`` we know
-            # for sure that it will be a `list`.
-            ids = ", ".join(e["create"]["_id"] for e in errors)  # type: ignore[union-attr]
-            msg = f"IDs '{ids}' already exist in the document store."
-            raise DuplicateDocumentError(msg)
+
+        if errors:
+            duplicate_errors_ids = []
+            other_errors = []
+            for e in errors:
+                if e["create"]["error"]["type"] == "version_conflict_engine_exception":
+                    if policy == DuplicatePolicy.FAIL:
+                        duplicate_errors_ids.append(e["create"]["_id"])
+                    elif policy == DuplicatePolicy.SKIP:
+                        # when the policy is skip, these errors are OK and we should not raise an exception
+                        continue
+                    else:
+                        other_errors.append(e)
+                else:
+                    other_errors.append(e)
```

Comment:
What do you think of this instead? Seems more straightforward to me. 🤔 


```suggestion
                error_type = e["create"]["error"]["type"]
                if policy == DuplicatePolicy.FAIL and error_type == "version_conflict_engine_exception":
                    duplicate_errors_ids.append(e["create"]["_id"])
                elif policy == DuplicatePolicy.SKIP and error_type == "version_conflict_engine_exception":
                    # when the policy is skip, duplication errors are OK and we should not raise an exception
                    continue
                else:
                    other_errors.append(e)
```

Reply:
To me too :smile: